scripts/bootstrap.py

```diff
@@ -0,0 +1,271 @@
+REST_HOST = '127.0.0.1'
+
+parser = argparse.ArgumentParser(
+    description='Configure a single-node jormungandr network')
+parser.add_argument('--rest-port', type=int, default=8443,
+                    help='The port for REST API to listen on')
+parser.add_argument('--bft', action='store_true',
+                    help='Use BFT consensus mode')
+parser.add_argument('--genesis-praos', action='store_true',
+                    help='Use Genesis-Praos consensus mode')
+parser.add_argument('--startup-script', action='store_true',
+                    help='Add startup script')
+parser.add_argument('--secret-path', metavar='SECRET_PATH',
+                    type=str, default=os.getcwd(), help='The path to write secret files')
+parser.add_argument('--config-path', type=str, default=os.getcwd(),
+                    help='The path to write config files')
+parser.add_argument('--data-path', type=str, default=os.getcwd(),
+                    help='The path to blockchain storage')
+parser.add_argument('--slot-duration', type=int, default=10,
+                    help='Duration of a slot in seconds')
+parser.add_argument('--slots-per-epoch', type=int,
+                    default=5000, help='Number of slots in each epoch')
+parser.add_argument('--no-colors', action='store_true',
+                    help='Disable shell colours in stdout')
+parser.add_argument('--overwrite', action='store_true',
+                    help='Overwrite existing blockchain storage files')
+parser.add_argument('--jcli', type=str, default='jcli', help='Path to jcli')
+parser.add_argument('--jormungandr', type=str,
+                    default='jormungandr', help='Path to jormungandr')
+
+args = parser.parse_args()
+
+if args.bft and args.genesis_praos:
+    raise RuntimeError('You must select one consensus to use')
+elif args.bft:
+    consensus = 'bft'
+elif args.genesis_praos:
+    consensus = 'genesis_praos'
+else:
+    raise RuntimeError(
+        'You must specify a consensus to use with --bft or --genesis-praos')
+
+jcli = args.jcli
+jormungandr = args.jormungandr
+
+if shutil.which(jormungandr) is None:
+    raise RuntimeError('{} not found'.format(jormungandr))
+if shutil.which(jcli) is None:
+    raise RuntimeError('{} not found'.format(jcli))
+
+config_path = Path(args.config_path)
+secret_path = Path(args.secret_path)
+data_path = Path(args.data_path)
+
+storage = Path(data_path / 'jormungandr-storage-test')
+
+try:
+    if os.listdir(str(storage)):
+        if args.overwrite:
+            shutil.rmtree(storage)
+        else:
+            print('error: directory {} contains blocks.sqlite already, use --overwrite to overwrite'.format(str(storage)))
+            sys.exit(1)
+except FileNotFoundError:
+    pass
+
+color_red = ''
+color_white = ''
+color_green = ''
+color_blue = ''
+
+if not args.no_colors:
+    try:
+        # colorama is often present in a system and provides cross-platform coloring
+        from colorama import init, Fore
```

Comment:
If we are about to add extra deps it may be work using some higher abstraction libs, we may or may not but at least you may be interested in:

[rich](https://github.com/willmcgugan/rich)
[typer](https://typer.tiangolo.com/)

Reply:
Not at all. I want to keep them optional and use the most common thing available. Many PyPi packages include `colorama` as a dependency, so I decided to go with it.

Now as I think of it, Windows 10 is said to [support ANSI codes](https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences), so we may try to go without introducing extra dependencies and see how it goes.

Reply:
Ok, no lib no (3rd) party :P